packages/dashboard/package.json

```diff
@@ -15,7 +15,7 @@
     "start:api": "api-server --port 50002 --trajectoryServerUrl ws://localhost:8006 --publicKey e2e/certs/keycloak.pub",
     "build": "react-scripts build",
     "build:storybook": "build-storybook -s public",
-    "test": "react-scripts test",
+    "test": "react-scripts test --env=jest-environment-jsdom-sixteen",
```

Comment:
> Added the `jest-environment-jsdom-sixteen` package and a test configuration for a `typescript - react-scripts - react testing library` bug explained here: https://stackoverflow.com/questions/61036156/react-typescript-testing-typeerror-mutationobserver-is-not-a-constructor

Sorry, I added this in the PR description, I'm realizing now that it was not the best place =)

Reply:
My node version is 12.20

Reply:
Does it work if we upgrade/downgrade CRA or testing-library?

Reply:
I'll create docker with different versions to check this

Reply:
CRA v3.4.3 -> v3.0.3
CRA 3.4.3 -> v4.0.1
RTL v11.0.4 -> v10.2.1
RTL v11.0.4  -> v11.2.2


Tests are working correctly upgrading and downgrading these versions. There is an error when running `npm run start` on the api-server when we upgrade  to CRA 4.0.1:

Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.  TS1205

```
     5 | import { WebSocketMiddleware } from './websocket-connect';
     6 | 
  >  7 | export { CustomLogger as Logger };
       |          ^
     8 | 
     9 | export interface RpcRequest<T = unknown> {
    10 |   version: string;
 
```

So I think we should upgrade versions of RTL and CRA on a FUP PR and fix that issue too.

Reply:
https://github.com/osrf/romi-dashboard/issues/178

Reply:
thanks, if I got it right, we shouldn't need to explicitly set the jest env anymore after we perform the upgrade? If possible I would like to avoid this as that may bring maintaince problems later on as jest or react-scripts updates its environment.

Reply:
Sure, I rebase this branch onto https://github.com/osrf/rmf-web/pull/180. Now we have the upgrade here and solved the problem.